### PR TITLE
Use json to format http logs

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #11224

```
23-11-22 11:42:00:314 [ info sync.httplogger ]:	REQUEST 33ea6fe2-2f0e-456a-ad78-811dca54003c {"request":{"body":{},"header":{"Accept":"*/*","Accept-Language":"en_CH","Authorization":"Basic [redacted]","OC-Connection-Validator":"desktop","Original-Request-ID":"33ea6fe2-2f0e-456a-ad78-811dca54003c","User-Agent":"Mozilla/5.0 (Windows) mirall/6.0.0-git (ownCloud, windows-10.0.22631 ClientArchitecture: x86_64 OsArchitecture: x86_64)","X-Request-ID":"33ea6fe2-2f0e-456a-ad78-811dca54003c"},"info":{"id":"33ea6fe2-2f0e-456a-ad78-811dca54003c","method":"GET","url":"http://demo.testlab.owncloud.works/owncloud/status.php"}}}
23-11-22 11:42:00:361 [ info sync.httplogger ]:	RESPONSE 33ea6fe2-2f0e-456a-ad78-811dca54003c {"response":{"body":{"data":"{\"installed\":true,\"maintenance\":false,\"needsDbUpgrade\":false,\"version\":\"10.13.2.3\",\"versionstring\":\"10.13.2\",\"edition\":\"Community\",\"productname\":\"ownCloud\",\"product\":\"ownCloud\"}","length":177},"header":{"Access-Control-Allow-Origin":"*","Authorization":"Basic [redacted]","Cache-Control":"no-store, no-cache, must-revalidate","Connection":"Keep-Alive","Content-Length":"177","Content-Security-Policy":"default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *","Content-Type":"application/json","Date":"Wed, 22 Nov 2023 10:41:51 GMT","Expires":"Thu, 19 Nov 1981 08:52:00 GMT","Keep-Alive":"timeout=5, max=100","Pragma":"no-cache","Server":"Apache/2.4.41 (Ubuntu)","Set-Cookie":"ocnuf8vi1fzr=siqm3qk59jnf916tsketeis1mu; path=/owncloud; HttpOnly; SameSite=Strict\noc_sessionPassphrase=qbBIkZjPqYiuhaArSDITcnKHw8p03ok3Hi8j9w70XPhLXO21nAtT2%2Fj%2FwKkVTqAyLX4iBgT7kW2xEqWWoUW64rF4b0PYcL5%2Bw1JKj2qW9zlulqiXkcM4pQLneL9leIl7; expires=Wed, 22-Nov-2023 11:01:51 GMT; Max-Age=1200; path=/owncloud; HttpOnly; SameSite=Strict","X-Content-Type-Options":"nosniff","X-Download-Options":"noopen","X-Frame-Options":"SAMEORIGIN","X-Permitted-Cross-Domain-Policies":"none","X-Robots-Tag":"none","X-XSS-Protection":"0"},"info":{"id":"33ea6fe2-2f0e-456a-ad78-811dca54003c","method":"GET","reply":{"cached":false,"duration":46,"durationString":"duration(0h, 0min, 0s, 46ms)","status":200,"version":"HTTP 2"},"url":"http://demo.testlab.owncloud.works/owncloud/status.php"}}}
```

### CODE REVIEW

File: src/libsync/httplogger.cpp [hunk 1]
```diff
@@ -42,47 +44,39 @@ bool isTextBody(const QString &s)
...
+
+    QJsonObject info{{QStringLiteral("method"), QString::fromUtf8(verb)}, {QStringLiteral("id"), QString::fromUtf8(id)}, {QStringLiteral("url"), url}};
+
     if (reply) {
-        stream << " " << reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() << " (";
+        QString durationString;
+        QDebug(&durationString).nospace() << duration;
+        QJsonObject replyInfo{{QStringLiteral("status"), reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt()},
+            {QStringLiteral("cached"), reply->attribute(QNetworkRequest::SourceIsFromCacheAttribute).toBool()},
+            // downcast to int, this is json
```
Comment (TheOneRing) [inline src/libsync/httplogger.cpp:67]: We could also use string? but int should be more than sufficient.

Comment (TheOneRing) [reply src/libsync/httplogger.cpp:67]: No but gcc complained, I guess because json does not support int64

Comment (fmoc) [reply src/libsync/httplogger.cpp:67]: There seems to be some weird 53-bit limit in the spec and various services seem to implement different workarounds (some actually send strings along with the downcast int values). This should be fine, it's rather unlikely we'll overflow.

File: src/libsync/httplogger.cpp [hunk 2]
```diff
@@ -121,14 +120,12 @@ void HttpLogger::logRequest(QNetworkReply *reply, QNetworkAccessManager::Operati
...
 
         const auto request = reply->request();
-        const auto keys = request.rawHeaderList();
-        QList<QNetworkReply::RawHeaderPair> header;
-        header.reserve(keys.size());
-        for (const auto &key : keys) {
-            header << qMakePair(key, request.rawHeader(key));
+        QJsonObject header;
+        for (const auto &key : request.rawHeaderList()) {
+            header[QString::fromUtf8(key)] = QString::fromUtf8(request.rawHeader(key));
```
Comment (fmoc) [inline src/libsync/httplogger.cpp:125]: Doesn't this overwrite the authorize header you set above?

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/libsync/httplogger.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/libsync/httplogger.cpp:125]: This is the header object before its passed to logHttp

**Review by fmoc — APPROVED** [review]

**Pull request merged**